### PR TITLE
minetest: depend on luajit-openresty

### PR DESCRIPTION
**Pull request by equal-l2**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by equal-l2** [comment]
This works on my Mac (macOS 12.0.1, x86) at least.

**Comment by equal-l2** [comment]
Relevant lines:
```
2021-11-08T23:20:46.4277950Z [1709/2316] /usr/bin/env EFL_RUN_IN_TREE=1 /private/tmp/efl-20211108-36049-u87dbz/efl-1.25.1/build/src/bin/edje/edje_cc -id /private/tmp/efl-20211108-36049-u87dbz/efl-1.25.1/data/elementary/themes/img -sd /private/tmp/efl-20211108-36049-u87dbz/efl-1.25.1/data/elementary/themes/snd ../src/bin/exactness/player_entry.edc src/bin/exactness/player_entry.edj
2021-11-08T23:20:46.4279470Z FAILED: src/bin/exactness/player_entry.edj 
2021-11-08T23:20:46.4281010Z /usr/bin/env EFL_RUN_IN_TREE=1 /private/tmp/efl-20211108-36049-u87dbz/efl-1.25.1/build/src/bin/edje/edje_cc -id /private/tmp/efl-20211108-36049-u87dbz/efl-1.25.1/data/elementary/themes/img -sd /private/tmp/efl-20211108-36049-u87dbz/efl-1.25.1/data/elementary/themes/snd ../src/bin/exactness/player_entry.edc src/bin/exactness/player_entry.edj

```

**Comment by equal-l2** [comment]
@cho-m Would you make the CI re-run so that I can see how this PR works on the other envs?

**Comment by cho-m** [comment]
@equal-l2 Please add a revision bump (i.e. `revision 1`) to your commit since this modifies the dependency tree.

Once you do this, it should automatically restart the CI.

**Comment by equal-l2** [comment]
Something is wrong on Linux:
```
make[2]: *** No rule to make target '/home/linuxbrew/.linuxbrew/opt/irrlicht/Frameworks/IrrFramework.framework', needed by 'bin/minetest'.  Stop.
```

**Comment by equal-l2** [comment]
The test fails on ARM.

**Comment by Calinou** [comment]
I can confirm this builds successfully on a 2020 Mac mini, but the resulting app can't be opened (I just get a "The application "minetest.app" can't be opened" message).

**Comment by Calinou** [comment]
Bump :slightly_smiling_face:

**Comment by SMillerDev** [comment]
The tests still fail on ARM, isn't this aimed to resolve that?

File: Formula/minetest.rb [hunk 1]
```diff
@@ -40,7 +47,10 @@ class Minetest < Formula
   depends_on "jpeg"
   depends_on "libogg"
   depends_on "libvorbis"
-  depends_on "luajit"
+  depends_on "luajit-openresty"
+  depends_on "openal-soft" => :on_linux
+
+  uses_from_macos "curl"
```
Comment (carlocab) [inline Formula/minetest.rb:53]:
```suggestion

  uses_from_macos "curl"

  on_linux do
    depends_on "openal-soft"
  end
```

File: Formula/minetest.rb [hunk 2]
```diff
@@ -49,12 +59,19 @@ def install
     args << "-DBUILD_CLIENT=1" << "-DBUILD_SERVER=0"
     args << "-DENABLE_FREETYPE=1" << "-DCMAKE_EXE_LINKER_FLAGS='-L#{Formula["freetype"].opt_lib}'"
     args << "-DENABLE_GETTEXT=1" << "-DCUSTOM_GETTEXT_PATH=#{Formula["gettext"].opt_prefix}"
-    args << "-DIRRLICHT_LIBRARY=#{Formula["irrlicht"].opt_frameworks}/IrrFramework.framework"
```
Comment (carlocab) [inline Formula/minetest.rb:52]: Is this not needed?

Comment (equal-l2) [reply Formula/minetest.rb:52]: That statement causes a build failure on Linux.
As other platforms work fine without the statement, I removed that.

Comment (carlocab) [reply Formula/minetest.rb:52]: How does this affect linkage with `irrlicht`?

Comment (equal-l2) [reply Formula/minetest.rb:52]: Do you mean how the build fails on Linux?

On Linux, the directory corresponding to `Formula["irrlicht"].opt_frameworks` is not generated and the compiler cannot find headers and link library.

Comment (carlocab) [reply Formula/minetest.rb:52]: No, I'm asking how the removal of this flag affects linkage with `irrlicht` on macOS.

Comment (equal-l2) [reply Formula/minetest.rb:52]: I didn't consider that.
Now I checked outputs of `brew linkage` with and without the statement.

<details>
<summary>With the statement:</summary>

```
System libraries:
  /usr/lib/libSystem.B.dylib
  /usr/lib/libbz2.1.0.dylib
  /usr/lib/libc++.1.dylib
  /usr/lib/libcurl.4.dylib
  /usr/lib/libform.5.4.dylib
  /usr/lib/libiconv.2.dylib
  /usr/lib/libncurses.5.4.dylib
  /usr/lib/libobjc.A.dylib
  /usr/lib/libsqlite3.dylib
  /usr/lib/libz.1.dylib
Variable-referenced libraries:
  @executable_path/../Frameworks/IrrFramework.framework/Versions/A/IrrFramework
  @executable_path/../Frameworks/libfreetype.6.dylib
  @executable_path/../Frameworks/libgmp.10.dylib
  @executable_path/../Frameworks/libjpeg.9.dylib
  @executable_path/../Frameworks/libluajit-5.1.2.dylib
  @executable_path/../Frameworks/libogg.0.dylib
  @executable_path/../Frameworks/libopenal.1.dylib
  @executable_path/../Frameworks/libpng16.16.dylib
  @executable_path/../Frameworks/libvorbis.0.dylib
  @executable_path/../Frameworks/libvorbisfile.3.dylib
Dependencies with no linkage:
  irrlicht
  libogg
  libvorbis
```
</details>

<details>
<summary>Without the statement:</summary>

```
System libraries:
  /usr/lib/libSystem.B.dylib
  /usr/lib/libbz2.1.0.dylib
  /usr/lib/libc++.1.dylib
  /usr/lib/libcurl.4.dylib
  /usr/lib/libform.5.4.dylib
  /usr/lib/libiconv.2.dylib
  /usr/lib/libncurses.5.4.dylib
  /usr/lib/libobjc.A.dylib
  /usr/lib/libsqlite3.dylib
  /usr/lib/libz.1.dylib
Variable-referenced libraries:
  @executable_path/../Frameworks/libfreetype.6.dylib
  @executable_path/../Frameworks/libgmp.10.dylib
  @executable_path/../Frameworks/libjpeg.9.dylib
  @executable_path/../Frameworks/libluajit-5.1.2.dylib
  @executable_path/../Frameworks/libogg.0.dylib
  @executable_path/../Frameworks/libopenal.1.dylib
  @executable_path/../Frameworks/libpng16.16.dylib
  @executable_path/../Frameworks/libvorbis.0.dylib
  @executable_path/../Frameworks/libvorbisfile.3.dylib
Dependencies with no linkage:
  irrlicht
  libogg
  libvorbis
```
</details>

Diff:
```diff
@@ -10,7 +10,6 @@
   /usr/lib/libsqlite3.dylib
   /usr/lib/libz.1.dylib
 Variable-referenced libraries:
-  @executable_path/../Frameworks/IrrFramework.framework/Versions/A/IrrFramework
   @executable_path/../Frameworks/libfreetype.6.dylib
   @executable_path/../Frameworks/libgmp.10.dylib
   @executable_path/../Frameworks/libjpeg.9.dylib
```

There is a difference in linkage indeed, though somewhat it doesn't affect the test.

**Comment by equal-l2** [comment]
minetest 5.5.0 has been released.
I'll make new PR for that.

**Pull request closed without merging**